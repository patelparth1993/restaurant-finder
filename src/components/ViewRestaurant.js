import React from "react";

import "./style/ViewRestaurant.scss";

export const ViewRestaurant = (props) => {
  let priceFilter = false;
  let resultCount = 0;
  if (+props.filterByPrice !== 0) {
    priceFilter = true;
  }
  return (
    <div data-testid='view-restaurant' className='view-restaurant-comp'>
      {props.resList.map((eachRes) => {
        let viewRes = false;
        viewRes =
          (eachRes.name.toLowerCase().indexOf(props.search) !== -1 ||
            eachRes.address.toLowerCase().indexOf(props.search) !== -1 ||
            eachRes.area.toLowerCase().indexOf(props.search) !== -1) &&
          (priceFilter ? eachRes.price === +props.filterByPrice : true);
        if (viewRes) {
          resultCount++;
          return (
            <div key={eachRes.id} className='restaurant-cards'>
              <strong>Name: </strong>
              {eachRes.name}
              <br />
              <strong>Address: </strong>
              {eachRes.address}
              <br />
              <strong>Area: </strong>
              {eachRes.area}
              <br />
              <strong>Price: </strong>
              {eachRes.price}
              <br />
            </div>
          );
        }
        return null;
      })}
      {resultCount === 0 && props.resList.length > 0 && (
        <div className='no-res-found'>No Restaurants Found</div>
      )}
    </div>
  );
};
ViewRestaurant.defaultProps = {
  resList: [],
  search: "",
  filterByPrice: 0,
};
