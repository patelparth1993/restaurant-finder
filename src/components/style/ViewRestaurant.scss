.view-restaurant-comp {
  display: flex;
  flex-wrap: wrap;
  margin-top: 20px;
  text-align: center;
  .restaurant-cards {
    width: 31%;
    padding: 10px;
    margin-bottom: 30px;
    border-radius: 10px;

    &:hover {
      background-color: gainsboro;
    }
  }
  .no-res-found {
    color: red;
    font-size: 20px;
  }

  @media only screen and (max-width: 600px) {
    display: block;
    .restaurant-cards {
      width: 100%;
    }
  }
}
