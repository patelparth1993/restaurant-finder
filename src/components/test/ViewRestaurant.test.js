import React from "react";
import "@testing-library/jest-dom";
import { render, cleanup } from "@testing-library/react";

import { ViewRestaurant } from "../ViewRestaurant";

afterEach(cleanup);

const fakeResList = [
  {
    id: 21307,
    name: "Scaramouche Restaurant",
    address: "1 Benvenuto Place",
    city: "Toronto",
    price: 4,
    area: "Toronto / SW Ontario",
  },
  {
    id: 82957,
    name: "The Sultan's Tent",
    address: "49 Front St E",
    city: "Toronto",
    price: 2,
    area: "Toronto / SW Ontario",
  },
];
it("should take a snapshot", () => {
  const { asFragment } = render(
    <ViewRestaurant resList={fakeResList} search={""} filterByPrice={0} />
  );

  expect(
    asFragment(
      <ViewRestaurant resList={fakeResList} search={""} filterByPrice={0} />
    )
  ).toMatchSnapshot();
});

describe("Restaurant component renders", () => {
  it("restaurant list", () => {
    const { getByTestId } = render(
      <ViewRestaurant resList={fakeResList} search={""} filterByPrice={0} />
    );
    const restaurantChildArray = Array.from(
      getByTestId("view-restaurant").children
    );

    expect(/.*Name.*Address.*Area.*$/.test(restaurantChildArray[0].textContent))
      .toBeTruthy;
  });

  it("No restaurants found if search is not met", () => {
    const { getByTestId } = render(
      <ViewRestaurant
        resList={fakeResList}
        search={"somesearch"}
        filterByPrice={0}
      />
    );
    expect(getByTestId("view-restaurant").textContent).toBe(
      "No Restaurants Found"
    );
  });
});
