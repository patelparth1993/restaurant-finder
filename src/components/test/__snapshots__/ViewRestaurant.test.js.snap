// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="view-restaurant-comp"
    data-testid="view-restaurant"
  >
    <div
      class="restaurant-cards"
    >
      <strong>
        Name: 
      </strong>
      Scaramouche Restaurant
      <br />
      <strong>
        Address: 
      </strong>
      1 Benvenuto Place
      <br />
      <strong>
        Area: 
      </strong>
      Toronto / SW Ontario
      <br />
      <strong>
        Price: 
      </strong>
      4
      <br />
    </div>
    <div
      class="restaurant-cards"
    >
      <strong>
        Name: 
      </strong>
      The Sultan's Tent
      <br />
      <strong>
        Address: 
      </strong>
      49 Front St E
      <br />
      <strong>
        Area: 
      </strong>
      Toronto / SW Ontario
      <br />
      <strong>
        Price: 
      </strong>
      2
      <br />
    </div>
  </div>
</DocumentFragment>
`;
